lib/optimizer/src/TransformationEngine.php

```diff
@@ -84,14 +84,14 @@ private function getTransformers()
             foreach ($this->configuration->get(Configuration::KEY_TRANSFORMERS) as $transformerClass) {
                 $arguments = [];
 
-                if (is_a($transformerClass, MakesRemoteRequests::class, true)) {
-                    $arguments[] = $this->remoteRequest;
-                }
-
                 if (is_a($transformerClass, Configurable::class, true)) {
                     $arguments[] = $this->configuration->getTransformerConfiguration($transformerClass);
                 }
 
+                if (is_a($transformerClass, MakesRemoteRequests::class, true)) {
+                    $arguments[] = $this->remoteRequest;
+                }
+
```

Comment:
Aside: instead of relying on positional arguments, would it be preferable to pass the arguments as an associative array of args so the order does not matter?

Reply:
I remember we had this discussion already. I would really try to stay away from the WordPress' "args pattern", as that makes dependency injection extremely difficult for anyone using it within a modern PHP framework, and makes it very liable to use for everyone else.

Just as an example for why I loathe it, this "args pattern" leads to bugs like this: https://github.com/ampproject/amp-wp/issues/4489

I'll change the code so that it uses reflection to detect the needed arguments instead. This is not something that happens thousands of times per request, so the performance impact of using reflection will be negligible.